### DISCUSSION TITLE
Wifi IP and hostname

### DISCUSSION THREAD

**Discussion by timaseth** [arendst/Tasmota#12232] [Ideas]
Hi All and special thanks to Adrian @ascillato2 for PR (#12091 ) which makes searching for a device IP on the network a thing of the past - nicely done, mate!

If I knew coding, I would have submitted a PR but might I offer a suggestion;
With reference to the above PR, just below the password input, we add the option to include a hostname too.
Some pesky browsers do not permit a fresh page to display correctly or even web page cache and history can trouble the new page from loading but armed with a hostname, a user may access the GUI without a worry!

Just a suggestion!

**Comment by ascillato2** [comment]
Thanks, Let me do some tests.

**Comment by mlavelle42** [comment]
Displaying the just-connected IP is indeed helpful. But this only seems to work with the initial connection. I run several routers and subsequent changes to a different router do not display the IP address. Displaying both the the IP address and hostname of the device each time a router change is made would be _very_ helpful.

**Reply by sfromis** [reply]
Using `SetOption53`, you can have the host name and IP address included in the header of the Tasmota web UI pages.

**Reply by mlavelle42** [reply]
I do that now and it is very helpful. The problem is that after changing to a different WiFi network, subsequent to the initial connection, there is no confirmation that the switch actually made it (the browser shows a "time out", which obviously means that the network switch occurred - but that's it), nor is there any indication of the new IP address. The only way to find the new IP is to search the just-connected-to network. That's not a problem if you know what you are doing but for a lot of people it's a non-starter ("now what do I do?").

**Reply by sfromis** [reply]
Easy solution, do not use the IP address, but instead host name. With a good setup, you can then just press refresh.

**Reply by sfromis** [reply]
The number of users switching between routers without being among those "you know what you are doing" is probably pretty low.

**Reply by mlavelle42** [reply]
Hmm... Yes, I've used hostnames in the past but the results are inconsistent. Just now, of the 4 routers I'm working with, only 2 properly handle hostnames. This is a big problem for the people I deal with. They easily get lost navigating different networks. I use WiFiConfig 4 with SSID2 set for a hotspot as the fallback. I do not want WiFiManager revealing devices are out there for exploiting. 

I think the current method, on initial configuration, of displaying the new IP address (and potentially displaying the hostname), and the fact that it switches my computer's WiFi over to the new network, is an excellent solution. This same technique would help immensely remove the ambiguity of what to do next after a device is connected to a different network.
